src/bgw/job.h

```diff
@@ -41,17 +42,16 @@ extern TimestampTz ts_bgw_job_timeout_at(BgwJob *job, TimestampTz start_time);
 
 extern TSDLLEXPORT bool ts_bgw_job_delete_by_id(int32 job_id);
 extern TSDLLEXPORT bool ts_bgw_job_update_by_id(int32 job_id, BgwJob *job);
-extern TSDLLEXPORT int32 ts_bgw_job_insert_relation(Name application_name,
-													Interval *schedule_interval,
-													Interval *max_runtime, int32 max_retries,
-													Interval *retry_period, Name proc_schema,
-													Name proc_name, Name owner, bool scheduled,
-													int32 hypertable_id, Jsonb *config);
+extern TSDLLEXPORT int32 ts_bgw_job_insert_relation(
+	Name application_name, Interval *schedule_interval, Interval *max_runtime, int32 max_retries,
+	Interval *retry_period, Name proc_schema, Name proc_name, Name check_schema, Name check_name,
+	Name owner, bool scheduled, int32 hypertable_id, Jsonb *config);
 extern TSDLLEXPORT void ts_bgw_job_permission_check(BgwJob *job);
 
 extern TSDLLEXPORT void ts_bgw_job_validate_job_owner(Oid owner);
 
 extern bool ts_bgw_job_execute(BgwJob *job);
```

Comment:
Should we run the config check before we execute the job, as well? This would allow us to centralize all the config validity checks in the check function. Otherwise we might run into an inconsistency at some point, where the check function and the execute function have different ideas about what is the valid config.

Reply:
Once registered, the config can only be changed through alter_job, or by modifying the table `_timescaledb_config.bgw_job`, right? The second case would lead to that kind of inconsistency but I don't think it's possible to get there if only using the APIs. But of course we can also check for config validity when executing